extensions/interactions/LogicProof/directives/LogicProof.js

```diff
@@ -91,7 +101,11 @@ oppia.directive('oppiaInteractiveLogicProof', [
           // NOTE: for information on integrating angular and code-mirror see
           // http://github.com/angular-ui/ui-codemirror
           $scope.codeEditor = function(editor) {
-            editor.setValue($scope.localQuestionData.default_proof_string);
+            if (!$scope.interactionIsActive) {
```

Comment:
Reduce the amount of stuff that's being branched on:

```
var proofString = (
  $scope.interactionIsActive ?
  $scope.localQuestionData.default_proof_string :
  $scope.getLastAnswer().proof_string));
editor.setValue(proofString);
```

Reply:
Done (changed everywhere)